cmd/cli/uninstall_mesh.go

```diff
@@ -199,6 +182,64 @@ func (d *uninstallMeshCmd) run() error {
 			fmt.Fprintf(d.out, "OSM namespace CANNOT be deleted in a managed environment\n")
 		}
 	}
+
+	err := d.deleteClusterResources()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *uninstallMeshCmd) findFlagSpecifiedMesh(meshInfoList []meshInfo) (bool, bool) {
+	meshFlagSpecified := d.meshName != ""
+	specifiedMeshFound := false
+	if meshFlagSpecified {
+		specifiedMeshFound = d.findMesh(meshInfoList)
+		if !specifiedMeshFound {
+			fmt.Fprintf(d.out, "Did not find mesh [%s] in namespace [%s]\n", d.meshName, d.meshNamespace)
+			// print a list of meshes within the cluster for a better user experience
+			err := d.printMeshes()
+			if err != nil {
+				fmt.Fprintf(d.out, "Unable to list meshes in the cluster - [%v]", err)
+			}
+		}
+	}
+	return meshFlagSpecified, specifiedMeshFound
+}
+
+func (d *uninstallMeshCmd) promptMeshUninstall(meshInfoList, meshesToUninstall []meshInfo, meshFlagSpecified bool) ([]meshInfo, error) {
+	for _, mesh := range meshInfoList {
+		// Only prompt for specified mesh if `mesh-name` is specified
+		if meshFlagSpecified && mesh.name != d.meshName {
+			continue
+		}
+		confirm, err := confirm(d.in, d.out, fmt.Sprintf("\nUninstall OSM [mesh name: %s] in namespace [%s] and/or OSM resources?", mesh.name, mesh.namespace), 3)
+		if err != nil {
+			return nil, err
+		}
+		if confirm {
+			meshesToUninstall = append(meshesToUninstall, mesh)
+		}
+	}
+	return meshesToUninstall, nil
+}
+
+func (d *uninstallMeshCmd) deleteNs(ctx context.Context, ns string) error {
+	if d.deleteNamespace {
```

Comment:
You can remove a level of indentation by reworking the check.

```suggestion
	if !d.deleteNamespace {
	    return nil
	}
```